tests/benchmark/s2n_negotiate_server_benchmark.cc

```diff
@@ -0,0 +1,483 @@
+    }
+
+    if (conn_settings.session_ticket || conn_settings.session_cache) {
+        /* Key initialization */
+        uint8_t *st_key;
+        uint32_t st_key_length;
+
+        if (session_ticket_key_file_path) {
+            int fd = open(session_ticket_key_file_path, O_RDONLY);
+            GUARD_EXIT(fd, "Error opening session ticket key file");
+
+            struct stat st;
+            GUARD_EXIT(fstat(fd, &st), "Error fstat-ing session ticket key file");
+
+            st_key = (uint8_t *) mmap(0, st.st_size, PROT_READ, MAP_PRIVATE, fd, 0);
+            st_key_length = st.st_size;
+
+            close(fd);
+        } else {
+            st_key = default_ticket_key;
+            st_key_length = sizeof(default_ticket_key);
+        }
+
+        if (s2n_config_add_ticket_crypto_key(config_once, ticket_key_name, strlen((char *) ticket_key_name), st_key,
+                                             st_key_length, 0) != 0) {
+            fprintf(stderr, "Error adding ticket key: '%s'\n", s2n_strerror(s2n_errno, "EN"));
+            exit(1);
+        }
+    }
+
+    bool stop_listen = false;
+    while ((!stop_listen) && (fd_bench = accept(sockfd, ai->ai_addr, &ai->ai_addrlen)) > 0) {
+        if (!parallelize) {
+            unsigned int len = sizeof(all_suites) / sizeof(all_suites[0]);
+            for (unsigned int j = 0; j < len; ++j) {
+                unsigned int suite_num = j;
+                if (num_user_certificates != num_user_private_keys) {
+                    fprintf(stderr, "Mismatched certificate(%d) and private key(%d) count!\n", num_user_certificates,
+                            num_user_private_keys);
+                    exit(1);
+                }
+
+                unsigned int num_certificates = 0;
+                if (num_user_certificates == 0) {
+                    if (all_suites[suite_num]->auth_method == S2N_AUTHENTICATION_RSA) {
+                        certificates[0] = rsa_certificate_chain;
+                        private_keys[0] = rsa_private_key;
+                        num_certificates = 1;
+                    } else {
+                        certificates[0] = ecdsa_certificate_chain;
+                        private_keys[0] = ecdsa_private_key;
+                        num_certificates = 1;
+                    }
+                } else {
+                    num_certificates = num_user_certificates;
+                }
+
+                for (unsigned int i = 0; i < num_certificates; i++) {
+                    struct s2n_cert_chain_and_key *chain_and_key = s2n_cert_chain_and_key_new();
+                    GUARD_EXIT(s2n_cert_chain_and_key_load_pem(chain_and_key, certificates[i], private_keys[i]),
+                               "Error getting certificate/key");
+
+                    GUARD_EXIT(s2n_config_add_cert_chain_and_key_to_store(config_once, chain_and_key),
+                               "Error setting certificate/key");
+                }
```

Comment:
I'm a little confused about the program flow here. I want to double check you're only loading the certificate once per cipher suite and not every iteration.